plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -611,8 +611,13 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
             // Parallel read-only trx execution enabled.
```

Comment:
For consistency, I think we could post read-only-trx to the read_only_trx_safe queue always. Instead of sometimes being executed via `push_transaction`.

Reply:
This is turning out to be a bit tricker than I anticipated. 
It requires the `process_read_only_transactions()` in start block. I still think it is cleaner than not piping read-only trxs through `push_transaction`, but I could see a desire to leave it where they flow through `push_transaction` if read-only thread pool is not enabled. Thoughts? @linh2931

Reply:
Oh no, I just realized we can not simply combine the two cases. For read-only trx with no threads, when you call `process_read_only_transactions`, you need to do set_db_read_only_mode and unset_db_read_only_mode, because those are done in window switching.

Reply:
Thanks. Fixed.

Reply:
I think the change should work. I'm thinking into it further and will let you if anything comes up.